src/Compilers/CSharp/Portable/Syntax/Syntax.xml

```diff
@@ -3560,12 +3644,18 @@
       </PropertyComment>
       <Kind Name="IdentifierToken"/>
     </Field>
-    <Field Name="AccessorList" Type="AccessorListSyntax" Override="true" Optional="true" />
-    <Field Name="ExpressionBody" Type="ArrowExpressionClauseSyntax" Optional="true" />
-    <Field Name="Initializer" Type="EqualsValueClauseSyntax" Optional="true"/>
-    <Field Name="SemicolonToken" Type="SyntaxToken" Optional="true">
-      <Kind Name="SemicolonToken" />
-    </Field>
+    <Choice>
+      <Field Name="AccessorList" Type="AccessorListSyntax" Override="true" />
+      <Sequence>
+        <Choice>
+          <Field Name="ExpressionBody" Type="ArrowExpressionClauseSyntax" />
+          <Field Name="Initializer" Type="EqualsValueClauseSyntax" />
+        </Choice>
+        <Field Name="SemicolonToken" Type="SyntaxToken">
+          <Kind Name="SemicolonToken" />
+        </Field>
+      </Sequence>
+    </Choice>
```

Comment:
this one is particularly nice.  It says:  You either need an accessor list.  Or you can have an `=> expr` or `= expr`, but both of those must be followed by a semicolon.  In g4 terms, it is:

```g4
property_declaration
  : attribute_list* modifier* type explicit_interface_specifier? identifier_token (accessor_list | ((arrow_expression_clause | equals_value_clause) ';'))
  ;
```